### ISSUE TITLE
something wrong in the file anchor_target_layer.py

### ISSUE THREAD

**Issue by LuNhChina** [jwyang/faster-rcnn.pytorch#256]
```
if cfg.TRAIN.RPN_POSITIVE_WEIGHT < 0:
            num_examples = torch.sum(labels[i] >= 0)
            positive_weights = 1.0 / num_examples
            negative_weights = 1.0 / num_examples
        else:
            assert ((cfg.TRAIN.RPN_POSITIVE_WEIGHT > 0) &
                    (cfg.TRAIN.RPN_POSITIVE_WEIGHT < 1))
```
In above code
1) in line 2, there is no definition for index i in previous
2) in the else branch, we didn't compute positive_weights and negative_weights

**Comment by z-huabao** [comment]
I think `i = batch_size - 1`, it seems no right.
in my idea, `num_examples = torch.sum(labels > 0, axis=1).mean()`

and `cfg.TRAIN.RPN_POSITIVE_WEIGHT` be -1 usually, so else branch no execute
